lib/puppetserver/ca/cli.rb

```diff
@@ -86,7 +87,15 @@ def self.run(cli_args = ARGV, out = STDOUT, err = STDERR)
           if exit_code
             return exit_code
           else
-            return action.run(input)
+            begin
+              return action.run(input)
+            rescue Puppetserver::Ca::Error => e
+              logger.err "Fatal error when running action '#{action_argument}'"
+              logger.err "  Arguments: #{general_options.merge(input)}"
```

Comment:
I'm a little on the fence about printing out the args here. in a lot of cases, it just seems unnecessarily verbose, because the invocation is usually right on the previous line.
```
puppetserver-ca-cli git:(server_list) ./exe/puppetserver-ca list
Fatal error when running action 'list'
  Arguments: {}
  Error: Failed connecting to https://localhost:8140/puppet-ca/v1/certificate_statuses/any_key
  Root cause: Failed to open TCP connection to localhost:8140 (Connection refused - connect(2) for "localhost" port 8140)
```
```
puppetserver-ca-cli git:(server_list) ./exe/puppetserver-ca list --all
Fatal error when running action 'list'
  Arguments: {"all"=>true}
  Error: Failed connecting to https://localhost:8140/puppet-ca/v1/certificate_statuses/any_key
  Root cause: Failed to open TCP connection to localhost:8140 (Connection refused - connect(2) for "localhost" port 8140)
```
Is there a particular thing you wanted to surface this way?

Reply:
I think it was just YAGNI

Reply:
Alright I'm in favor of removing it then.